### PR TITLE
Add Precompiles Page

### PR DESCRIPTION
**Pull request by danyalprout**
**Description**

As part of Fjord we've added a precompile for [RIP-7212](https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md). This is documented in the [Fjord overview](https://github.com/ethereum-optimism/specs/blob/main/specs/fjord/overview.md), but it maybe useful to have a centralized precompiles page, similar to how we have a predeploys and preinstalls page.

### CODE REVIEW

File: specs/glossary.md [hunk 1]
```diff
@@ -149,6 +150,16 @@ available to improve the L2's UX.
 
 All preinstall contracts are specified in the [preinstalls specification](./protocol/preinstalls.md).
 
+## Precompiled Contract ("Precompile")
+
+[predeploy]: glossary.md#precompiled-contract-precompile
```
Comment (sebastianst) [inline specs/glossary.md]:
```suggestion
[precompile]: glossary.md#precompiled-contract-precompile
```

File: specs/protocol/precompiles.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+OP-Stack chains contain the [standard Ethereum precompiles](https://www.evm.codes/precompiled) as well as a small
+number of additional precompiles. The following table lists each of the additional precompiles. The system version
+indicates when the precompile was introduced.
+
+| Name       | Address                                    | Introduced |
+|------------| ------------------------------------------ |------------|
+| P256VERIFY | 0x0000000000000000000000000000000000000100 | Fjord      |
+
+## P256VERIFY
```
Comment (sebastianst) [inline specs/protocol/precompiles.md:27]: I think it's also worth backlinking to this section from the Fjord overview page where RIP-7212 is listed as a feature.

File: specs/glossary.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
   - [Chain Re-Organization](#chain-re-organization)
   - [Predeployed Contract ("Predeploy")](#predeployed-contract-predeploy)
   - [Preinstalled Contract ("Preinstall")](#preinstalled-contract-preinstall)
+  - [Precompiled Contract ("Precompile")](#precompiled-contract-precompile)
```
Comment (sebastianst) [inline specs/glossary.md:16]: I would actually list it before Predeploys. Then we go from most to less native.

File: specs/SUMMARY.md [hunk 4]
```diff
@@ -20,6 +20,7 @@
     - [Batch Submitter](./protocol/batcher.md)
   - [Predeploys](./protocol/predeploys.md)
   - [Preinstalls](./protocol/preinstalls.md)
+  - [Precompiles](./protocol/precompiles.md)
```
Comment (sebastianst) [inline specs/SUMMARY.md]: Same, I would list it before Predeploys. Then we go from most to less native.

**Review by sebastianst — CHANGES_REQUESTED** [review]
Thanks!

File: specs/protocol/precompiles.md (hunk 2 above)
Comment (danyalprout) [reply specs/protocol/precompiles.md:27]: I updated the Fjord overview page to link here, which has a link to the RIP. If you think it's worth keeping both the precompile and RIP link on the overview page -- let me know.

**Review by sebastianst — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**